pyparsing/results.py

```diff
@@ -623,27 +628,25 @@ def pprint(self, *args, **kwargs):
     # add support for pickle protocol
     def __getstate__(self):
         return (
-            self.__toklist,
+            self._toklist,
             (
-                self.__tokdict.copy(),
-                self.__parent is not None and self.__parent() or None,
-                self.__accumNames,
-                self.__name,
+                self._tokdict.copy(),
+                self._parent is not None and self._parent() or None,
+                self._all_names,
+                self._name,
             ),
         )
 
     def __setstate__(self, state):
-        self.__toklist = state[0]
-        self.__tokdict, par, inAccumNames, self.__name = state[1]
-        self.__accumNames = {}
-        self.__accumNames.update(inAccumNames)
+        self._toklist, (self._tokdict, par, inAccumNames, self._name) = state
+        self._all_names = set(inAccumNames)
         if par is not None:
-            self.__parent = wkref(par)
+            self._parent = wkref(par)
         else:
-            self.__parent = None
+            self._parent = None
 
     def __getnewargs__(self):
-        return self.__toklist, self.__name, self.__asList, self.__modal
+        return self._toklist, self._name, self._modal
```

Comment:
Why is aslist arg removed here?

Reply:
Good catch.  I now removed this function.  It is used by pickle to create new ParseResults objects before their attributes are set.  Since no constructor code should be running when Pickle is recreating these objects, sending no parameters to `__new__` is the correct solution.

That said, this function is never run during testing.

Reply:
I will remove this function in another PR